examples/llama/model_test.py

```diff
@@ -141,12 +144,16 @@ def test_module(
     params_dict = module_ark.params_dict()
     param_names = set(params_dict.keys())
 
-    if os.path.exists(pth_path):
+    rank = torch.distributed.get_rank()
+    world_size = torch.distributed.get_world_size()
+    checkpoints = sorted(Path(ckpt_dir).glob("*.pth"))
+    ckpt_path = checkpoints[rank]
+    if os.path.exists(ckpt_path):
         prefix = module_name_prefix + "." if module_name_prefix else ""
         # Load the state_dict from the given path
-        state_dict_pt = torch.load(pth_path)
+        state_dict_pt = torch.load(ckpt_path)
         state_dict_pt = {
-            k[len(prefix) :]: v
+            k[len(prefix) :]: v.to(dtype=numpy_dtype_to_torch_dtype[dtype])
```

Comment:
Can you try [`torch.set_default_dtype`](https://pytorch.org/docs/stable/generated/torch.set_default_dtype.html#torch-set-default-dtype) instead? I think this is the official way to do this.